lib/mix/tasks/inertia.install.ex

```diff
@@ -0,0 +1,312 @@
+  def update_esbuild_config(igniter) do
+    igniter
+    |> Igniter.Project.Config.configure(
+      "config.exs",
+      :esbuild,
+      [:version],
+      "0.21.5"
+    )
+    |> Igniter.Project.Config.configure(
+      "config.exs",
+      :esbuild,
+      [Igniter.Project.Application.app_name(igniter)],
+      {:code,
+       Sourceror.parse_string!("""
+       [
+        args:
+          ~w(js/app.jsx --bundle --chunk-names=chunks/[name]-[hash] --splitting --format=esm  --target=es2020 --outdir=../priv/static/assets --external:/fonts/* --external:/images/*),
+        cd: Path.expand("../assets", __DIR__),
+        env: %{"NODE_PATH" => Path.expand("../deps", __DIR__)}
+       ]
+       """)}
+    )
+    |> Igniter.add_task("esbuild.install")
+  end
+
+  @doc false
+  def setup_client(igniter) do
+    case igniter.args.options[:client_framework] do
+      "react" ->
+        igniter
+        |> install_client_package()
+        |> Igniter.create_new_file("assets/js/app.jsx", inertia_app_jsx(), on_exists: :overwrite)
+
+      framework when framework in ["vue", "svelte"] ->
+        igniter
+        |> install_client_package()
+        |> Igniter.assign(:client_framework, framework)
+
+      _ ->
+        igniter
+    end
+  end
+
+  defp install_client_package(igniter) do
+    packages =
+      case igniter.args.options[:client_framework] do
+        "react" -> "@inertiajs/react react react-dom"
+        "vue" -> "@inertiajs/vue3 vue vue-loader"
+        "svelte" -> "@inertiajs/svelte svelte"
+      end
+
+    igniter
+    |> Igniter.add_task("cmd", ["npm install --prefix assets #{packages}"])
+  end
+
+  defp inertia_app_jsx() do
+    """
+    import React from "react";
+    import axios from "axios";
+
+    import { createInertiaApp } from "@inertiajs/react";
+    import { createRoot } from "react-dom/client";
+
+    axios.defaults.xsrfHeaderName = "x-csrf-token";
+
+    createInertiaApp({
+      resolve: async (name) => {
+        return await import(`./pages/${name}.jsx`);
+      },
+      setup({ App, el, props }) {
+        createRoot(el).render(<App {...props} />);
+      },
+    });
+    """
+  end
+
+  defp print_next_steps(igniter) do
+    client_framework = igniter.assigns[:client_framework]
+
+    next_steps = []
+
+    next_steps =
+      if client_framework do
+        client_setup_steps = """
+        To finish setting up the client side integration:
+        1. Create your Inertia pages in your assets/pages directory
+        2. Set up your entry point file to initialize Inertia and default layout
+        """
+
+        next_steps ++ [client_setup_steps]
+      else
+        next_steps
+      end
+
+    next_steps =
+      next_steps ++
+        [
+          """
+          For more information on using Inertia with Phoenix, refer to:
+          https://hexdocs.pm/inertia/readme.html
+          """
+        ]
+
+    # Add completion notice to the Igniter
+    Enum.each(next_steps, &Igniter.add_notice(igniter, &1))
```

Comment:
You'd need to `Enum.reduce/1` this is discarding the result.

Reply:
Good catch - fixed 👍